rio_tiler/tasks.py

```diff
@@ -1,26 +1,29 @@
 """rio_tiler.tasks: tools for handling rio-tiler's future tasks."""
 
 from concurrent import futures
-from typing import (
-    Any,
-    Callable,
-    Dict,
-    Generator,
-    Iterator,
-    Optional,
-    Sequence,
-    Tuple,
-    Union,
-)
+from typing import Any, Callable, Dict, Generator, Optional, Sequence, Tuple, Union
 
 import numpy
 
 from .constants import MAX_THREADS
 from .logger import logger
 
-TaskType = Union[
-    Generator[Tuple[Callable, str], None, None], Iterator[Tuple[futures.Future, str]],
-]
+
+class LikeFuture:
```

Comment:
I'm open to suggestions for a better class name. 

as the name tell, this Object is like a future, meaning that it has a `.result` method which will be in a non-threading env to run the function and get the results